packages/kit/test/lib-HookSystem.types.ts

```diff
@@ -0,0 +1,56 @@
+import { HookSystem } from "../src/lib";
+
+// Placeholder function
+const fn = () => null;
+
+// Hook system hooks
+type Hooks = {
+	hook1: () => null;
+	hook2: { fn: () => null; meta: { foo: string } };
+};
+
+// System
+const system = new HookSystem<Hooks>();
+
+// Scoped variant
+const { hook } = system.createScope("_", []);
+
+/**
+ * Hooks requiring no meta don't require a meta argument.
+ */
+system.hook("_", "hook1", fn);
+hook("hook1", fn);
+
+/**
+ * Hooks requiring no meta accept an optional meta argument.
```

Comment:
What's the purpose of accepting an optional meta argument?

Reply:
None ^^'